compiler/src/dotty/tools/dotc/core/CompilationUnitInfo.scala

```diff
@@ -1,21 +1,33 @@
 package dotty.tools.dotc.core
 
 import dotty.tools.io.AbstractFile
+import dotty.tools.tasty.TastyVersion
 
 /** Information about the compilation unit of a class symbol.
   *
   * @param associatedFile The source or class file from which this class or
   *                       the class containing this symbol was generated,
   *                       null if not applicable.
+  * @param tastyVersion   The TASTy version (major, minor, experimental)
   */
 class CompilationUnitInfo(
   val associatedFile: AbstractFile,
+  private var tastyVersionOpt: Option[TastyVersion],
```

Comment:
also can this be nullable? or we have a special "empty" tasty version - to save on allocation of `Some`(or cache the Some?)

Reply:
I used option to make it simpler to use where we need it. I added `ClassSymbol. tastyVersion: Option[TastyVersion]` to avoid handling the nulability of `compilationUnitInfo`

https://github.com/lampepfl/dotty/pull/19010/files#diff-ebd0ee134453eb7cd2794dfcdbaab7b9c6b9cc4ba722d7ee03f6a87100959067R286

If I do no make it an option have we would need to create a new one each time we check `ClassSymbol. tastyVersion`.

We should probably cache the TASTy version.